### ISSUE TITLE
How can i get the attention weight for each layer on the encoder?

### ISSUE THREAD

**Issue by kimwin2** [facebookresearch/detr#492]
Nowadays i'm trying to improve detr performance on the side of the weakly supervised learning. 
So i want to use the attention weight for each layer on the encoder. 
Can you recommand any way to get that?
In my guess, i'm focusing on the transformer.py like as bellow:

TransformerEncoderLayer(nn.Module)/ def forward_post/
    q = k = self.with_pos_embed(src, pos)
        src2 = self.self_attn(q, k, value=src, attn_mask=src_mask,
                              key_padding_mask=src_key_padding_mask)[0]
        src = src + self.dropout1(src2)
        src = self.norm1(src)
        src2 = self.linear2(self.dropout(self.activation(self.linear1(src))))
        src = src + self.dropout2(src2)
        src = self.norm2(src)
        return src

Maybe src2 can be a attention weight. but I don't know how to visualize it.
Can you recommand any way to visualize the attention weight?

Thank you very much and i appreciate your effort.
